src/test/java/org/javarosa/core/model/data/test/GeoPointDataTests.java

```diff
@@ -20,4 +33,39 @@ public void getDisplayText_whenAllComponentsAreZero_returnsEmptyString() {
 
         assertThat(zeroedOut.getDisplayText(), is(""));
     }
+
+    @Test
+    public void getDisplayText_whenAccuracyOmitted_HasThreeComponents() {
+        GeoPointData point = new GeoPointData(new double[]{2.3, 7.3, 3.2});
+
+        assertThat(point.getDisplayText(), is("2.3 7.3 3.2"));
+    }
+
+    @Test
+    public void missingAccuracy_isNotTreatedAs0() throws IOException, XFormParser.ParseException {
+        Scenario scenario = Scenario.init("Missing accuracy", html(
+            head(
+                title("Missing accuracy"),
+                model(
+                    mainInstance(
+                        t("data id=\"missing-accuracy\"",
+                            t("q1"),
+                            t("accuracy_rounded"),
+                            t("note")
+                        )),
+                    bind("/data/q1").type("geopoint"),
+                    bind("/data/accuracy_rounded").calculate("round(selected-at(/data/q1, 3), 2)"),
+                    bind("/data/note").relevant("/data/accuracy_rounded = 0"))),
+            body(
+                input("/data/q1"),
+                input("/data/note")
+            )
+        ));
+
+        scenario.answer("/data/q1","1.234 5.678");
+        assertThat(scenario.getAnswerNode("/data/note").isRelevant(), is(false));
+
+        scenario.answer("/data/q1", "1.234 5.678 0 0");
```

Comment:
Suggestion:

- Having a test where the altitude is not zero: `1.234 5.678 0 0`
- Having another test where values can produce NaN (e.g. then value is nullable, field is not relevant): `ABC 23.55, PP 0`

Reply:
@lognaturel, would you prefer to leave this suggestion for another PR?